GitUI/UserControls/RevisionGrid/Graph/RevisionGraphRevision.cs

```diff
@@ -89,9 +89,9 @@ public int EnsureScoreIsAbove(int minimalScore)
 
         public ObjectId Objectid { get; set; }
 
-        public ConcurrentBag<RevisionGraphRevision> Parents { get; }
-        public ConcurrentBag<RevisionGraphRevision> Children { get; }
-        public SynchronizedCollection<RevisionGraphSegment> StartSegments { get; }
+        public ThreadSafeGrowingList<RevisionGraphRevision> Parents { get; }
```

Comment:
So I guess we don't need Parents/Children to behave like a set, then?

Reply:
No, the parents are added as they are received from the git log output. I think we can rely on this information to be correct.